### PR TITLE
add drop column, rename column/index for incompatible ddl changes

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb-tools/issues/596 https://github.com/pingcap/tiflow/issues/9850

### What is changed and how it works?
- add `drop column`
- add `rename column`
- add `rename index`
- regard DropDatabase,DropTable,DropIndex,RenameTable,TruncateTable,DropSchema,DropTablePartition,TruncateTablePartition as  incompatible ddls

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Review by okJiang — APPROVED** [review]

File: pkg/binlog-filter/filter.go [hunk 1]
```diff
@@ -81,23 +81,23 @@ const (
...
-	Rename                 EventType = "rename"
-	Drop                   EventType = "drop"
-	Truncate               EventType = "truncate"
-	ModifyPK               EventType = "modify pk"
-	ModifyUK               EventType = "modify uk"
+	RenameColumn           EventType = "rename column"
+	RenameIndex            EventType = "rename index"
+	DropColumn             EventType = "drop column"
+	DropPK                 EventType = "drop pk"
+	DropUK                 EventType = "drop uk"
```
Comment (lichunzhu) [inline pkg/binlog-filter/filter.go]: How about
```suggestion
	DropPrimaryKey                 EventType = "drop primary key"
	DropUniqueKey                 EventType = "drop unique key"
```

**Review by lichunzhu — APPROVED** [review]
Rest LGTM

**Review by lichunzhu — APPROVED** [review]

**Pull request merged**